### PR TITLE
merge multiplication counter test

### PR DESCRIPTION
**Pull request by bpurinton**
This is my first attempt at a merge into the course material, and my first time requesting code review. So bear with me.

## Problem

I wanted to patch the `string_multiplication.rb` test to prevent it from passing with the code

```
p "Ho" + "Ho" + "Ho"
```

## Solution

I made a counter, looped through the file, and searched for the presence of `*`, with the condition that it doesn't occur on a commented line. So as long as there is one `*` in their code somewhere, it passes. 

## Why I request review

I'm not sure if my solution is good and logical, or if it could be improved. Also, there are still other ways I could imagine cheating the test like:

```
variable = 5 * 6
p "Ho" + "Ho" + "Ho"
```

but I'm not sure we need to worry about every single edge case.

### CODE REVIEW

File: spec/scripts/string_spec.rb [hunk 1]
```diff
@@ -8,6 +8,18 @@
...
+    multiplication_counter = 0
+    File.foreach(multiplication_file).with_index do |line, line_num|
+      if line.include?("*") 
+        unless line.include?("#")
+          multiplication_counter += 1
+        end
+      end
+    end
+    expect(multiplication_counter).to be >= 1,
+      "Expected 'string_addition.rb' to use the * multiplication method, but only #{multiplication_counter} '*' found."
```
Comment (jelaniwoods) [inline spec/scripts/string_spec.rb]: I wonder if this error message can be improved at all 🤔 This message will only really show up if they haven't used `*` at all, which means `multiplication_counter` is always `0`. I feel like the way the message is phrased, it sounds like `multiplication_counter` could be another number and still be wrong, when that's not really the case.

> Expected 'string_addition.rb' to use the * multiplication method, but only 0 '*' found

Maybe the word "only" feels a little off to me here. What do you think about removing that word?

**Review by jelaniwoods — APPROVED** [review]
@bpurinton Thanks for the PR!

> but I'm not sure we need to worry about every single edge case.

I agree, we don't need to address every edge case in these tests.

I had one minor suggestion about the error message, but you can take it or leave it.

LGTM, you can merge when ready.

File: spec/scripts/string_spec.rb (hunk 1 above)
Comment (bpurinton) [reply spec/scripts/string_spec.rb]: I agree. Let's drop the only. Thanks for the feedback.

**Comment by bpurinton** [comment]
Dropped only. Ready to merge the PR.
